lib/board.ex

```diff
@@ -24,46 +24,51 @@ defmodule Board do
 
   @spec move_status(map(), integer()) :: atom()
   def move_status(board, player_move) do
-    if is_out_of_bounds?(board, player_move) do
-      :out_of_bounds
-    else
-      :valid
+    cond do
+      is_out_of_bounds?(board, player_move) -> :out_of_bounds
+      is_space_taken?(board, player_move) -> :space_taken
+      true -> :valid
     end
   end
 
+  @spec is_space_taken?(map(), integer()) :: boolean()
+  defp is_space_taken?(board, player_move) do
+    Map.fetch!(board, player_move) != :empty
+  end
+
   @spec is_out_of_bounds?(map(), integer()) :: boolean()
   defp is_out_of_bounds?(board, player_move) do
     player_move not in 1..Enum.count(board)
   end
 
   @spec has_player_won?(map()) :: boolean()
   defp has_player_won?(board) do
-    Enum.any?([get_all_rows_status(board), get_all_columns_status(board), get_both_diagonals_status(board)])
+    Enum.any?([all_rows_status(board), all_columns_status(board), both_diagonals_status(board)])
   end
 
   @spec is_board_full?(map) :: boolean
   defp is_board_full?(board) do
     !Enum.member?(Map.values(board), :empty)
   end
 
-  @spec get_board_dimensions(map()) :: integer()
-  def get_board_dimensions(board) do
+  @spec board_dimensions(map()) :: integer()
+  def board_dimensions(board) do
     trunc(:math.sqrt(Enum.count(board)))
   end
 
-  @spec get_all_rows_status(map()) :: boolean()
-  defp get_all_rows_status(board) do
-    board_dimensions = get_board_dimensions(board)
+  @spec all_rows_status(map()) :: boolean()
```

Comment:
I'm not entirely sure what this name implies until I look at the implementation. It looks like it's determining if the game is won when all the tokens are the same in a row? Maybe we can call it `won_by_roll?` or something similar? Also it's a [convention](https://hexdocs.pm/elixir/main/naming-conventions.html#trailing-question-mark-foo) in Elixir that a function that return a boolean has a trailing question mark.

Reply:
This is a really great point, I completely agree that (especially after an iteration of name changing for these functions) the name is not incredibly clear as to describe what is happening, `win_by_row?` or something similar with the question mark makes a lot more sense to me and I think is a great idea and the way to go! When I made the current name I was trying to avoid using the word win as I didn't know about the trailing questions mark naming convention, but for some reason after learning about the naming convention where a function that returns a boolean has either a trailing question mark or a leading `is` or both, I forgot to refactor these functions to include it like some of the other functions in board, thank you for the wake up call! 😄